torchx/specs/api.py

```diff
@@ -61,6 +62,7 @@ class Resource:
     cpu: int
     gpu: int
     memMB: int
+    devices: list[Tuple[str, int]] = field(default_factory=list)
```

Comment:
Any reason for the list? Using a map makes it so keys are forced to be unique -- we probably don't ever want someone to specify the same resource key twice